### PR TITLE
feat(#861): add optional parameter --from to set the start point

### LINKED ISSUE #861: Add start-point option to `git feature`
Currently, `git feature` creates a new branch based on the current active one. 

As `git flow` specifies that feature branches should be based on `develop`, it would be nice to be able to specify a `start-point`, so we don't need to switch branches first.

After finishing a feature, I currently need to run 3 commands to create my new feature branch:

```bash
git checkout develop
git pull
git feature my-feat
```

It would be awesome if we could simplify those 3 lines to a single command:

```bash
# git feature <branchname> [<start-point>]
  git feature my-feat develop 
```

Regardless of my active branch, `git feature my-feat develop` would:

- pull/fetch the commits from `origin`
- create a new branch with `develop` as `start-point`, 
- check out the new feature branch.

Does `git-extras` have global options? It would be awesome to be able to set the default `start-point` trough a global option

### PR DESCRIPTION
**Pull request by vanpipy**
Close #861

### CODE REVIEW

File: bin/git-create-branch [hunk 1]
```diff
@@ -20,6 +20,13 @@ do
 				REMOTE=origin
 			fi
 			;;
+		--from)
+			if [[ -n $2 ]]
+			then
```
Comment (SuperSandro2000) [inline bin/git-create-branch:25]:
```suggestion
			if [[ -n $2 ]]; then
```

File: bin/git-create-branch [hunk 2]
```diff
@@ -44,9 +51,16 @@ then
 	rm -f "$stderr"
 	if [ $REMOTE_EXIT -eq 0 ]
 	then
-		git push $REMOTE HEAD:refs/heads/$BRANCH
-		git fetch $REMOTE
-		git checkout --track -b $BRANCH $REMOTE/$BRANCH
+		if [[ -n $START_POINT ]]
+		then
```
Comment (SuperSandro2000) [inline bin/git-create-branch:55]:
```suggestion
		if [[ -n $START_POINT ]]; then
```

**Review by SuperSandro2000 — COMMENTED** [review]
you get the gist.

File: bin/git-create-branch (hunk 1 above)
Comment (vanpipy) [reply bin/git-create-branch]: Update.

File: bin/git-create-branch (hunk 2 above)
Comment (vanpipy) [reply bin/git-create-branch]: Update.

And a question I want to ask is how to choose the type of using the style of the `if then`:

Style A:
```
if [[ ... ]]; then
   ...
fi
```

Style B:
```
if [[ ... ]]
then
    ...
fi
```

Both of them exist in the file. That confuse me a little.

Thanks :D.

**Comment by spacewander** [comment]
@vanpipy 
Would you update the doc?

**Comment by vanpipy** [comment]
> @vanpipy
> Would you update the doc?

Hi, @spacewander ,

The git-create-branch and git-feature docs will update at the next commit.

**Review by spacewander — APPROVED** [review]

**Pull request merged**